frame/support/src/storage/migration.rs

```diff
@@ -389,6 +389,39 @@ pub fn move_prefix(from_prefix: &[u8], to_prefix: &[u8]) {
 	}
 }
 
+/// A phony migration that does nothing, except executing `TryDecodeEntireStorage` on
+/// `post_upgrade`, which implies it is only available if `try-state` feature is used.
+///
+/// This can be used typically in the top level runtime, where `AllPallets` typically comes from
+/// `construct_runtime!`.
+pub struct EnsureStateDecodes<AllPallets>(sp_std::marker::PhantomData<AllPallets>);
+
+#[cfg(not(feature = "try-runtime"))]
+impl<AllPallets> crate::traits::OnRuntimeUpgrade for EnsureStateDecodes<AllPallets> {
+	fn on_runtime_upgrade() -> crate::weights::Weight {
+		Default::default()
```

Comment:
Why did you add this empty impl? Why not emit an error when someone tries to use `EnsureStateDecodes` with pallets that dont support it?

Reply:
> use EnsureStateDecodes with pallets that dont support it? 

not sure what this means? in `not(feature="try-runtime")` this code still will be executed.

Reply:
Oh right. Do you dont have to remove it for the real run, yes did not see that.